### ISSUE TITLE
Move "Box Primitive" to "Box" component docs

### ISSUE THREAD

**Issue by sebald** [marigold-ui/marigold#2748]
## Description

Currently we have two pages which are concerned with the `<Box>`. This makes it harder to find all the documentation

## Proposal

Remove the "Box Primitive" page and incorporate its contents into the `Box` component documentation

https://marigold-ui.io/concepts/box-primitive/ -> https://marigold-ui.io/components/box/

Also, after #2747 is done, remove the style props documentation! If this was not already done.
